src/lib/tzcode/CMakeLists.txt

```diff
@@ -1,2 +1,53 @@
-add_library(tzcode STATIC strftime.c strptime.c timelocal.c timezone.c)
+add_library(tzcode STATIC
+            strftime.c strptime.c timelocal.c localtime.c timezone.c)
 target_link_libraries(tzcode)
+
+set(TZCODE_PROJECT tzcode-github)
+set(TZCODE_BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR})
+set(TZCODE_PREFIX_DIR tz)
+set(TZCODE_PREFIX ${TZCODE_BUILD_DIR}/${TZCODE_PREFIX_DIR})
+set(TZDATA_TARGETS main.zi)
+set(TZCODE_SRC ${PROJECT_SOURCE_DIR}/src/lib/tzcode)
+
+list(TRANSFORM TZDATA_TARGETS
+     PREPEND ${TZCODE_PREFIX}/src/${TZCODE_PROJECT}/
+     OUTPUT_VARIABLE TZDATA_FULL_TARGETS)
+
+ExternalProject_Add(${TZCODE_PROJECT}
+    GIT_REPOSITORY https://github.com/eggert/tz
+    GIT_TAG 2022a
+    GIT_SHALLOW YES
+
+    EXCLUDE_FROM_ALL 1
+    INSTALL_DIR ${TZCODE_BUILD_DIR}/install
+    PREFIX ${TZCODE_PREFIX_DIR}
+
+    UPDATE_COMMAND ""
+    CONFIGURE_COMMAND ""
+
+    BUILD_COMMAND ${CMAKE_COMMAND} -E env
+        CC=${CMAKE_C_COMPILER}
+        CFLAGS=${CMAKE_C_FLAGS}
+        PACKRATDATA=backzone
+        make -C <SOURCE_DIR> ${TZDATA_TARGETS}
+    BUILD_IN_SOURCE TRUE
```

Comment:
Because it makes no much sense to try to build out-of-source, given the fact that external library sources will be downloaded and extracted inside of build directory under `/src/lib/tzcode/tzcode-github-prefix/src/tzcode-github/`.